app/open.js

```diff
@@ -34,22 +37,37 @@ define(['app/database', 'jquery', 'bootstrap', 'xlsx', 'parsley', 'bootstrap-sel
 			return false; // Don't submit form
 		})
 		.on('form:success', function() {
-			console.log("done");
-			process_wb(JSON.parse(saveData));
+			// Save data into the database
+			var name = $('#uploadForm').find('input[name="name"]').val();
+			var language = $('#uploadForm').find('select[name="language"]').val();
+			var subject = $('#uploadForm').find('select[name="subject"]').val();
+			var currentdate = new Date();
+
+			db.executeQuery("addDataset", [0, name, language, subject, 0, 0, date.formatDate(currentdate), date.formatDate(currentdate)]);
+			var id = db.lastInsertRowId("tbldatasets", "dataset_id");
+			// Save all items in the dataset
+			process_data(JSON.parse(saveData), id);
+			db.close();
 			window.location = 'index.html';
 		});
 	});
 
-	window.Parsley.addValidator('fileXlsx', {
-		validateString: function(_value, maxSize, parsleyInstance) {
-			return correctUpload;
-		},
-		requirementType: 'integer',
-		messages: {
-			en: 'This file is not supported.'
-		}
-	});
+	// Function for saving all items in the dataset
+	function process_data(data,name) {
```

Comment:
Rename this function. As you have to add a comment to both it's definition and its use, its name is not clear enough.